### PR TITLE
Comment Updates

### PR DESCRIPTION
**Pull request by Yorkemartin**
comment updates for a number of libs

### CODE REVIEW

File: src/entities/trade.ts [hunk 1]
```diff
@@ -110,8 +122,11 @@ export class Trade<TInput extends Currency, TOutput extends Currency, TTradeType
 
   /**
    * Constructs an exact in trade with the given amount in and route
-   * @param route route of the exact in trade
-   * @param amountIn the amount being passed in
+   * @template TInput The input token, either Ether or an ERC-20
+   * @template TOutput The output token, either Ether or an ERC-20
+   * @param route The route of the exact in trade
+   * @param amountIn The amount being passed in
+   * @returns The amount In trade
```
Comment (willpote) [inline src/entities/trade.ts:129]: in*

File: src/nonfungiblePositionManager.ts [hunk 2]
```diff
@@ -73,7 +73,11 @@ export type IncreaseOptions = CommonAddLiquidityOptions & IncreaseSpecificOption
 
 export type AddLiquidityOptions = MintOptions | IncreaseOptions
 
-// type guard
+/**
+ * Type guard
+ * @param options
+ * @returns mint
+ */
```
Comment (willpote) [inline src/nonfungiblePositionManager.ts:80]: Doesn't seem useful, recommend removing

File: src/selfPermit.ts [hunk 3]
```diff
@@ -21,7 +21,11 @@ export interface AllowedPermitArguments {
 
 export type PermitOptions = StandardPermitArguments | AllowedPermitArguments
 
-// type guard
+/**
+ * type guard
+ * @param permitOptions
+ * @returns Allowed permit
+ */
```
Comment (willpote) [inline src/selfPermit.ts:28]: Same

File: src/utils/calldata.ts [hunk 4]
```diff
@@ -15,6 +15,11 @@ export interface MethodParameters {
   value: string
 }
 
+/**
+ * To hex
```
Comment (willpote) [inline src/utils/calldata.ts:19]: Converts a big int to hex string

File: src/utils/maxLiquidityForAmounts.ts [hunk 5]
```diff
@@ -2,6 +2,13 @@ import { BigintIsh } from '@uniswap/sdk-core'
 import JSBI from 'jsbi'
 import { Q96 } from '../internalConstants'
 
+/**
+ * Returns an imprecise maximum amount of liquidity received for a given amount of token 0
```
Comment (willpote) [inline src/utils/maxLiquidityForAmounts.ts:6]: Would be great to explain what makes this imprecise (I don't know myself and expect other devs wouldn't know either)

File: src/utils/maxLiquidityForAmounts.ts [hunk 6]
```diff
@@ -10,6 +17,13 @@ function maxLiquidityForAmount0Imprecise(sqrtRatioAX96: JSBI, sqrtRatioBX96: JSB
   return JSBI.divide(JSBI.multiply(JSBI.BigInt(amount0), intermediate), JSBI.subtract(sqrtRatioBX96, sqrtRatioAX96))
 }
 
+/**
+ * Returns a precise maximum amount of liquidity received for a given amount of token 0
```
Comment (willpote) [inline src/utils/maxLiquidityForAmounts.ts:21]: Same

**Review by willpote — CHANGES_REQUESTED** [review]

**Review by Yodag7354 — APPROVED** [review]

File: src/utils/maxLiquidityForAmounts.ts [hunk 7]
```diff
@@ -2,6 +2,17 @@ import { BigintIsh } from '@uniswap/sdk-core'
...
 
+/**
+ * Returns an imprecise maximum amount of liquidity received for a given amount of token 0.
+ * This function is available to accommodate LiquidityAmounts#getLiquidityForAmount0 in the v3 periphery,
+ * which could be more precise by at least 32 bits by dividing by Q64 instead of Q96 in the intermediate step,
+ * and shifting the subtracted ratio left by 32 bits. This imprecise calculation will likely be replaced in a future
+ * v3 router contract.
+ * @param sqrtRatioAX96 price at lower boundary
+ * @param sqrtRatioBX96 price at upper boundary
+ * @param amount0 token0 amount
```
Comment (willpote) [inline src/utils/maxLiquidityForAmounts.ts:13]: Missing "The". Here and below

File: src/utils/maxLiquidityForAmounts.ts [hunk 8]
```diff
@@ -21,6 +40,13 @@ function maxLiquidityForAmount0Precise(sqrtRatioAX96: JSBI, sqrtRatioBX96: JSBI,
   return JSBI.divide(numerator, denominator)
 }
 
+/**
+ * Computes the maximum amount of liquidity received for a given amount of token1
+ * @param sqrtRatioAX96
+ * @param sqrtRatioBX96
+ * @param amount1
```
Comment (willpote) [inline src/utils/maxLiquidityForAmounts.ts:47]: Missing descriptions

File: src/entities/trade.ts [hunk 9]
```diff
@@ -14,7 +14,15 @@ import { ONE, ZERO } from '../internalConstants'
...
 
-// extension of the input output comparator that also considers other dimensions of the trade in ranking them
+/**
+ * Trades comparator, an extension of the input output comparator that also considers other dimensions of the trade in ranking them
+ * @template TInput The input token, either Ether or an ERC-20
+ * @template TOutput The output token, either Ether or an ERC-20
+ * @template TTradeType The trade type, either exact input or exact output
+ * @param a The first trade to compare
+ * @param b The second trade to compare
+ * @returns The
```
Comment (willpote) [inline src/entities/trade.ts:24]: Missing returns

**Review by willpote — APPROVED** [review]

**Comment by Yodag7354** [comment]
Update

**Pull request merged**